### PR TITLE
A rudimentary support for Scala 3

### PR DESCRIPTION
**Pull request by azolotko**

### CODE REVIEW

**Comment by unkarjedy** [comment]
related https://youtrack.jetbrains.com/issue/SCL-19086

**Comment by azolotko** [comment]
@unkarjedy might be also related to https://github.com/sbt/sbt/issues/6628

**Review by mutcianm — APPROVED** [review]

File: ideaSupport/src/main/scala/org/jetbrains/sbtidea/Quirks.scala [hunk 1]
```diff
@@ -12,19 +12,24 @@ trait Quirks { this: Keys.type =>
...
 
   def filterScalaLibraryCp(cp: Classpath): Classpath =
-    cp.filterNot(_.get(moduleID.key).exists(_.name.contains("scala-library")))
+    cp.filterNot(_.get(moduleID.key).exists(isScalaLibrary))
 
   def hasPluginsWithScala(plugins: Seq[IntellijPlugin]): Boolean =
     plugins.exists(plugin => pluginsWithScala.exists(id => plugin.toString.matches(s".*$id.*")))
+
+  private def isScalaLibrary(moduleId: ModuleID): Boolean = moduleId.name match {
+    case "scala-library" | "scala3-library_3" => true
```
Comment (jastice) [inline ideaSupport/src/main/scala/org/jetbrains/sbtidea/Quirks.scala:32]: is this confirmed to work with Scala 2.x? Don't they contain some version additionally?

**Comment by izzyreal** [comment]
Can someone comment on the status of this PR?

@jastice I suppose we're awaiting @azolotko's reply to https://github.com/JetBrains/sbt-idea-plugin/pull/111#discussion_r992417083 ?

**Comment by jastice** [comment]
> Can someone comment on the status of this PR?
> 
> @jastice I suppose we're awaiting @azolotko's reply to [#111 (comment)](https://github.com/JetBrains/sbt-idea-plugin/pull/111#discussion_r992417083) ?

Yes, though of course you can check it yourself :)

**Comment by izzyreal** [comment]
> > Can someone comment on the status of this PR?
> > @jastice I suppose we're awaiting @azolotko's reply to [#111 (comment)](https://github.com/JetBrains/sbt-idea-plugin/pull/111#discussion_r992417083) ?
> 
> Yes, though of course you can check it yourself :)

Yeah I was thinking the same 😄  👍  I primarily wanted to confirm that this PR is still relevant. Thanks!

**Comment by azolotko** [comment]
> Can someone comment on the status of this PR?
> 
> @jastice I suppose we're awaiting @azolotko's reply to [#111 (comment)](https://github.com/JetBrains/sbt-idea-plugin/pull/111#discussion_r992417083) ?

Oops 😬

I’ll give it another try tomorrow.

**Comment by izzyreal** [comment]
> > Can someone comment on the status of this PR?
> > @jastice I suppose we're awaiting @azolotko's reply to [#111 (comment)](https://github.com/JetBrains/sbt-idea-plugin/pull/111#discussion_r992417083) ?
> 
> Oops 😬
> 
> I’ll give it another try tomorrow.

That would be great, but I'm also happy to look into it. I'll keep an eye out on this PR. Thanks in advance!

**Comment by azolotko** [comment]
Ok, I've rebased the PR and given it another try.

[Here's a sample project ](https://github.com/azolotko/sbt-idea-example-scala3) that uses [a locally built snapshot](https://github.com/azolotko/sbt-idea-example-scala3/blob/scala3/project/plugins.sbt#L1) of sbt-ide-plugin from [the azolotko:rudimentary-scala-3-support branch](https://github.com/azolotko/sbt-idea-plugin/commit/24399fff). The project uses [Scala 3.3.0](https://github.com/azolotko/sbt-idea-example-scala3/blob/scala3/build.sbt#L8) and accesses both [Scala 2 and Scala 3 standard library classes](https://github.com/azolotko/sbt-idea-example-scala3/blob/scala3/src/main/scala/org/jetbrains/scala/samples/ui/MyToolWindowFactory.scala#L13-L14).

From the test run log:
```
2023-09-01 18:40:46,784 [  15818]   INFO - STDOUT - Illegal command line: hello from Scala 3 std lib
2023-09-01 18:40:46,784 [  15818]   INFO - STDERR - hello from Scala 2 std lib 
```

LGTM

**Comment by unkarjedy** [comment]
@azolotko 
Could you please also describe in words which issue this PR tries to fix?
I don't recall any existing issues related to lack of Scala 3 support, which would be related to sbt-idea-plugin.

**Comment by azolotko** [comment]
When I initially created this PR, the problem was in sbt-idea-plugin not
bundling Scala 3’s std lib, only Scala 2’s one.

Op vr 1 sep. 2023 om 19:09 schreef Dmitrii Naumenko <
***@***.***>

**Comment by unkarjedy** [comment]
@azolotko Thanks for the PR, even though it took so long to merge it!

**Comment by unkarjedy** [comment]
It's merged and published in 3.26.0

**Pull request closed without merging**